### ISSUE TITLE
REST API default parameter value

### ISSUE THREAD

**Issue by chalucha** [rejectedsoftware/vibe.d#1230]
I've run into a problem with default values.

Code:
```d
interface ITestAPI {
	string postDefault(int value, bool check = true);
}

class Test : ITestAPI {
	string postDefault(int value, bool check = true) {
		import std.format;
		return format("Value: %s, Check: %s", value, check);
	}
}

shared static this() {
	auto routes = new URLRouter;
	registerRestInterface(routes, new Test());

	auto settings = new HTTPServerSettings();
	settings.port = 8080;
	settings.bindAddresses = ["::1", "127.0.0.1"];

	listenHTTP(settings, routes);
}
```

If tested with:
```curl -i -X POST -d '{"value":42}' -H "Content-Type: application/json"  "http://127.0.0.1:8080/default"```

It returns: "Value: 42, Check: **true**" -- **OK**

```curl -i -X POST -d '{"value":42,"check":true}' -H "Content-Type: application/json"  "http://127.0.0.1:8080/default"```

It returns: "Value: 42, Check: **false**" -- **WRONG**
It returns false event with "check":1 and "check":"true"

Am I missing something or is this a bug?

**Comment by Geod24** [comment]
Could you check if it works with the automatic client ?
Also, which compiler / arch / version of Vibe.d ?

**Comment by chalucha** [comment]
With added
```d
runTask({
		auto client = new RestInterfaceClient!ITestAPI("http://127.0.0.1:8080/");
		auto res = client.postDefault(42, true);
		logInfo("Result: %s", res);
	});
```

Output: Result: Value: 42, Check: **false**

So the same error.

Compiler: dmd-2.068 (but same with ldc2, debug/release, x86/x86_64)
Arch: linux x86_64
Vibe current master, also tried with 0.7.24

**Comment by chalucha** [comment]
Ah, forgot to add that it's the same also on Win 8.1 x32_64 with dmd-2.068 and vibe-0.7.24

**Comment by Geod24** [comment]
Thanks for reporting !
It seems that the branch taken when the parameter is not explicitly set has a bug in it.
You can work around it by using `@bodyParam("check", "check")`.
Does it affect a lot of code, or is the workaround enough ? I expect it to be fixed once #1209 lands in.

**Comment by chalucha** [comment]
Thanks, I'm ok with workaround and waiting for #1209. There is no point to fix something probably fixed already in that PR.

**Comment by s-ludwig** [comment]
The fix should be in 0.7.25 and now there is also a regression test.
